src/sim/TrackInitData.hh

```diff
@@ -254,12 +255,11 @@ void resize(
     // Allocate device data
     auto capacity = params.capacity;
     make_builder(&data->initializers.storage).resize(capacity);
-    make_builder(&data->parents.storage).resize(capacity);
+    make_builder(&data->parents).resize(size);
     make_builder(&data->secondary_counts).resize(size);
 
-    // Start with an empty vector of track initializers and parent thread IDs
+    // Start with an empty vector of track initializers
```

Comment:
The function is called `resize` but seems to act more like a `reserveAndClear` (see for example the comment at line 264), and and is behaving differently from its STL "homophone" when doing:
```
builder->resize( N );
// do stuff
buider->resize( N+1 );
// stuff (at least some of it) not preserved
```